tensorflow/compiler/xla/service/gpu/tests/gemm_rewrite_test.cc

```diff
@@ -4658,6 +4658,162 @@ TEST_F(CublasLtF8GemmRewriteTest, ScaledABScaledDWithDAmaxF8) {
+                d_is_col ? "{0,1}" : "{1,0}";
+            combinations[i++] = std::array{
+                lcd, rcd, a_shape, b_shape, a_layout, b_layout, output_layout};
+          }
+        }
+      }
+    }
+  }
+  const char* hlo_template = R"(
+      HloModule test
+    ENTRY test {
+      x = f8e4m3fn<<Ashape>><<Alayout>> parameter(0)
+      x_f32 = f32<<Ashape>><<Alayout>> convert(x)
+      x_scale = f32[] parameter(2)
+      x_scale_bcast = f32<<Ashape>> broadcast(x_scale), dimensions={}
+      x_unscaled = f32<<Ashape>> multiply(x_f32, x_scale_bcast)
+      y = f8e4m3fn<<Bshape>><<Blayout>> parameter(1)
+      y_f32 = f32<<Bshape>><<Blayout>> convert(y)
+      y_scale = f32[] parameter(3)
+      y_scale_bcast = f32<<Bshape>> broadcast(y_scale), dimensions={}
+      y_unscaled = f32<<Bshape>> multiply(y_f32, y_scale_bcast)
+      ROOT out = f32[64,16]<<Olayout>> dot(x_unscaled, y_unscaled), lhs_contracting_dims=<<Lcd>>, rhs_contracting_dims=<<Rcd>>
+    }
+      )";
+  for (const auto& combination : combinations) {
+    absl::flat_hash_map<absl::string_view, absl::string_view> replacements;
+    replacements["<<Lcd>>"] = std::get<0>(combination);
+    replacements["<<Rcd>>"] = std::get<1>(combination);
+    replacements["<<Ashape>>"] = std::get<2>(combination);
+    replacements["<<Bshape>>"] = std::get<3>(combination);
+    replacements["<<Alayout>>"] = std::get<4>(combination);
+    replacements["<<Blayout>>"] = std::get<5>(combination);
+    replacements["<<Olayout>>"] = std::get<6>(combination);
+    const auto hlo_text = absl::StrReplaceAll(hlo_template, replacements);
+    EXPECT_TRUE(RunAndCompare(hlo_text, ErrorSpec{1e-4, 0.}));
+
+    MatchOptimizedHlo(hlo_text,
+                      R"(
+    ; CHECK:           custom_call_target="__cublas$lt$matmul$f8",
+          )");
+  }
+};
+
+TEST_F(CublasLtF8GemmRewriteTest, ScaledABUnscaledDF8ParameterizedBatched) {
+  // TODO(wenscarl): For batched matmaul, not all combinations of A, B and
+  // output layouts get pattern matched successfully to FP8 custom call. Only
+  // a handful of cases are tested here.
+  if (!GetCudaComputeCapability().IsAtLeast(
+          se::CudaComputeCapability::HOPPER)) {
+    GTEST_SKIP()
+        << "cuBLASLt FP8 kernels require Hopper or newer architecture.";
+  }
+
+  auto assemble_layout = [](std::vector<int>& layout_v) {
+    return "{" +
+           std::accumulate(++layout_v.begin(), layout_v.end(),
+                           std::to_string(layout_v[0]),
+                           [](const std::string& a, int b) {
+                             return a + "," + std::to_string(b);
+                           }) +
+           "}";
+  };
+  std::array<std::array<std::string, 7>, 384> combinations;
+  std::string lcd, rcd, a_shape, b_shape, a_layout, b_layout, o_layout;
+  int i = 0;
+  for (bool o_is_col : {false, true}) {
+    for (bool a_is_col : {false, true}) {
+      for (bool b_is_col : {false, true}) {
```

Comment:
Both `a_is_col` and `b_is_col` are unused.
Should they be removed or used somewhere in this test?

Reply:
Good catch! Since I want this fix in quickly, I'll fix in the internal CL

Reply:
Yes. Should be removed.

Reply:
As @reedwm proposed, I made the fix manually and landed the PR internally. Apparently the PR and the Google-internal patch went out of sync, potentially because the PR has been renamed and further modified recently. Hence, this Github PR has not been marked as merged automatically.
I hope I did not create too much of a mess here. Please let me know if manually closing this PR and sending the remaining code changed separately is an acceptable way forward.

Reply:
Hi Thomas, thank you very much for your efforts to get this PR in. No qualms on our side as to the method of operation that achieves that purpose!

Reply:
With regard to [0e7e616](https://github.com/tensorflow/tensorflow/commit/0e7e6165b04b3b22f1251309407964c6ddce208f), we decided it's better to restore original tests where file checking new optimized HLO instead of __cublas$lt$matmul$f8 generally. Those tests will be added back in PR [59807](https://github.com/tensorflow/tensorflow/pull/59804). Sorry about the back-and-forth.